### PR TITLE
fix: enhance upload custom progress bar

### PR DESCRIPTION
**Pull request by morenyang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

- https://github.com/ant-design/ant-design/issues/24213
- https://github.com/ant-design/ant-design/pull/24319

### 💡 Background and solution

- fix progress bar `type` could be overridden
- update demo

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix progress bar type in upload could be overridden |
| 🇨🇳 Chinese |   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/upload/demo/customize-progress-bar.md](https://github.com/morenyang/ant-design/blob/enhance-upload-custom-progress-bar/components/upload/demo/customize-progress-bar.md)

### CODE REVIEW

File: components/upload/demo/customize-progress-bar.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
 ---
 order: 999
 title:
-  zh-CN: Customize Progress Bar
+  zh-CN: 自定义进度条样式
   en-US: Customize Progress Bar
 debug: true
```
Comment (xrkffgg) [inline components/upload/demo/customize-progress-bar.md:6]:
```suggestion

```

File: components/upload/demo/customize-progress-bar.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
 ---
 order: 999
```
Comment (xrkffgg) [inline components/upload/demo/customize-progress-bar.md:2]: 看下现有的顺序，排到最后一个。

File: components/upload/UploadList.tsx [hunk 3]
```diff
@@ -177,7 +177,7 @@ export default class UploadList extends React.Component<UploadListProps, any> {
         // show loading icon if upload progress listener is disabled
         const loadingProgress =
           'percent' in file ? (
-            <Progress type="line" {...progressProps} percent={file.percent} />
+            <Progress {...progressProps} type="line" percent={file.percent} />
```
Comment (xrkffgg) [inline components/upload/UploadList.tsx:180]: 在 API 上注明一下吧，只支持 line

Comment (afc163) [reply components/upload/UploadList.tsx:180]: progressProps 的 ts 定义也可以把 type 和 percent 都 Omit 掉。

Comment (morenyang) [reply components/upload/UploadList.tsx:180]: > progressProps 的 ts 定义也可以把 type 和 percent 都 Omit 掉。

already done in 

https://github.com/ant-design/ant-design/blob/ac50fa147cd2425b890f252972df632c2e871b05/components/upload/interface.tsx#L71

File: components/upload/index.zh-CN.md [hunk 4]
```diff
@@ -44,7 +44,7 @@ title: Upload
 | onDownload | 点击下载文件时的回调，如果没有指定，则默认跳转到文件 url 对应的标签页。 | Function(file): void | 跳转新标签页 |  |
 | transformFile   | 在上传之前转换文件。支持返回一个 Promise 对象   | Function(file): `string | Blob | File | Promise<string | Blob | File>` | 无   |  |
 | iconRender | 自定义显示 icon | (file: UploadFile, listType?: UploadListType) => React.ReactNode | 无 |  |
-| progress | 自定义进度条样式 | [progress props](/components/progress/) | `{ strokeWidth: 2, showInfo: false }` |  |
+| progress | 自定义进度条样式 | [progress props](/components/progress/) ( 除 `type` 和 `percent` ) | { strokeWidth: 2, showInfo: false } |  |
```
Comment (xrkffgg) [inline components/upload/index.zh-CN.md:47]:
```suggestion
| progress | 自定义进度条样式 | [ProgressProps](/components/progress/#API) ( 仅支持 `type="line"` ) | `{ strokeWidth: 2, showInfo: false }` | 4.3.0 |
```

File: components/upload/UploadList.tsx (hunk 3 above)
Comment (afc163) [reply components/upload/UploadList.tsx:180]: cool

**Review by xrkffgg — APPROVED** [review]

**Pull request merged**